src/lib/Repository/UiLocationService.php

```diff
@@ -52,14 +63,38 @@ public function loadLocations(ContentInfo $contentInfo)
         return $uiLocations;
     }
 
+    /**
+     * Deletes locations.
+     * Throws exception if there is an attempt to delete the main location.
+     *
+     * @param array $locationIds
+     * @param int $contentId
+     *
+     * @throws ForbiddenException
+     */
+    public function deleteLocations(array $locationIds, int $contentId)
+    {
+        $contentInfo = $this->contentService->loadContentInfo($contentId);
+
+        foreach ($locationIds as $locationId) {
+            $location = $this->locationService->loadLocation($locationId);
+
+            if ($this->isMainLocation($location, $contentInfo)) {
+                throw new ForbiddenException('Main location cannot be deleted.');
+            }
+
+            $this->locationService->deleteLocation($location);
```

Comment:
Good point, I suppose these will be more likely handled in the controller than in the service but need some way of displaying this to the user depending upon the approach used. 
Happy to leave this as it is for now 😄